aiohttp/helpers.py

```diff
@@ -373,10 +401,22 @@ def content_disposition_header(disptype: str,
             if not key or not (TOKEN > set(key)):
                 raise ValueError('bad content disposition parameter'
                                  ' {!r}={!r}'.format(key, val))
-            qval = quote(val, '') if quote_fields else val
-            lparams.append((key, '"%s"' % qval))
-            if key == 'filename':
-                lparams.append(('filename*', "utf-8''" + qval))
```

Comment:
I'm not sure this is correct. `filename*` must be added if `filename` is not ASCII-only per RFC 6266, appendix D. The logic is supposed to be roughly https://github.com/cherrypy/cherrypy/pull/1851/files#diff-9d544371f1a0e2079e68ac3cca04bc6cd1ccaf92797f378b285f307871b8d6f0R32-R53. Note that NFKC is supposed to be used for normalization.

Reply:
Looking closer, it seems you may be right about the newer RFC...

Reply:
Hi. As this is the payload and not http Content-Disposition header, `filename*` can not be used. I believe your link was about the http header.

Reply:
Ah, then I misread what you're doing here.